docs/integrations/rsyslog/index.rst

```diff
@@ -0,0 +1,238 @@
+       -d papertrail -t rsyslog \
+       -c server=logsN.papertrailapp.com -c port=XXXXX \
+       -c format=rfc3164 -c tls=true
+
+loggly
+~~~~~~
+
+In addition to the server and port you also need a *customer token*
+which you then **need** to give as part of the ``-c sd`` parameter when
+creating the endpoint.
+
+::
+
+   avn service integration-endpoint-create --project your-project \
+       -d loggly -t rsyslog \
+       -c server=logs-01.loggly.com -c port=514 \
+       -c format=rfc5424 -c tls=true \
+       -c sd='TOKEN@NNNNN TAG="tag-of-your-choice"'
+
+sumo logic
+~~~~~~~~~~
+
+You need to the give the collector token as the ``-c sd`` parameter and
+use the server and port of the collector.
+
+::
+
+   avn service integration-endpoint-create --project your-project \
+       -d loggly -t rsyslog \
+       -c server=syslog.collection.XX.sumologic.com -c port=6514 \
+       -c tls=true -c format=rfc5424 \
+       -c sd='collector-token-string@NNNNN'
+
+Datadog
+~~~~~~~
+
+For Datadog integration you need to use custom format with logline
+
+::
+
+   avn service integration-endpoint-create --project your-project \
+       -d datadog -t rsyslog \
+       -c server=intake.logs.datadoghq.com -c port=10516 \
+       -c tls=true -c format=custom \
+       -c logline='DATADOG_API_KEY <%pri%>1 %timestamp:::date-rfc3339% %HOSTNAME% %app-name% - - - %msg%'
+
+.. Note::
+   If you want to use Datadog EU environment, the service address is:
+   
+   * ``server``: ``tcp-intake.logs.datadoghq.eu``
+   * ``port``: ``443``
+
+:doc:`Further details on send metrics and logs to Datadog </docs/integrations/datadog/index>`
+
+
+NewRelic
+~~~~~~~~
+
+You will also need a custom logline format for NewRelic Syslog
+integration. This is so you can prepend your `NewRelic Insights Insert
+Key <https://docs.newrelic.com/docs/apis/intro-apis/new-relic-api-keys/>`__
+and ensure the format matches the `built-in Grok
+pattern <https://docs.newrelic.com/docs/logs/ui-data/built-log-parsing-rules/#syslog-rfc5424>`__
+.
+
+::
+
+   avn service integration-endpoint-create --project your-project \
+   -d newrelic -t rsyslog \
+   -c server=newrelic.syslog.nr-data.net -c port=6514 \
+   -c tls=true -c format=custom \
+   -c logline='NEWRELIC_INSIGHTS_INSERT_KEY <%pri%>%protocol-version% %timestamp:::date-rfc3339% %hostname% %app-name% %procid% %msgid% -  %msg%'
+
+Coralogix
+~~~~~~~~~
+
+For coralogix integration you need to use custom format with logline
+
+::
+
+   avn service integration-endpoint-create --project your-project \
+   -d coralogix -t rsyslog \
+   -c server=syslogserver.coralogix.us -c port=5142 \
+   -c tls=false -c format=custom \
+   -c logline="{\"fields\": {\"private_key\":\"YOUR_CORALOGIX_KEY\",\"company_id\":\"YOUR_COMPANY_ID\",\"app_name\":\"%app-name%\",\"subsystem_name\":\"programname\"},\"message\": {\"message\":\"%msg%\",\"program_name\":\"programname\",\"pri_text\":\"%pri%\",\"hostname\":\"%HOSTNAME%\"}}"
```

Comment:
Is `programname` meant to be replaced by specific text?

Reply:
Nice catch, it should have been a variable `% programname%`